spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/MessageChannelConfigurerTests.java

```diff
@@ -0,0 +1,119 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.config;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.binder.redis.config.RedisMessageChannelBinderConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.junit.redis.RedisTestSupport;
+import org.springframework.cloud.stream.tuple.Tuple;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.Assert;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration({MessageChannelConfigurerTests.TestSource.class,
+		MessageChannelConfigurerTests.TestSink.class})
+public class MessageChannelConfigurerTests {
+
+	@Rule
+	public RedisTestSupport redisTestSupport = new RedisTestSupport();
+
+	@Autowired @Bindings(TestSink.class)
+	private Sink testSink;
+
+	@Autowired
+	@Bindings(TestSource.class)
+	private Source testSource;
+
+	@Test
+	public void testContentTypeConfigurer() throws Exception {
+		final CountDownLatch latch = new CountDownLatch(1);
+		testSink.input().subscribe(new MessageHandler() {
+			@Override
+			public void handleMessage(Message<?> message) throws MessagingException {
+				Assert.isTrue(message.getPayload() instanceof Tuple);
+				Assert.isTrue(((Tuple)message.getPayload()).getFieldNames().get(0).equals("message"));
+				Assert.isTrue(((Tuple) message.getPayload()).getValue(0).equals("Hi"));
+				latch.countDown();
+			}
+		});
+		testSource.output().send(MessageBuilder.withPayload("{\"message\":\"Hi\"}").build());
+		latch.await(10, TimeUnit.SECONDS);
```

Comment:
When I run these tests in STS, this message goes to the other test....

       : Channel 'bootstrap:-1.input' has 2 subscriber(s).